db/hybrid_logical_vector_test.go

```diff
@@ -229,3 +229,51 @@ func TestHybridLogicalVectorPersistence(t *testing.T) {
 	assert.Equal(t, inMemoryHLV.PreviousVersions, hlvFromPersistance.PreviousVersions)
 	assert.Equal(t, inMemoryHLV.MergeVersions, hlvFromPersistance.MergeVersions)
 }
+
+func TestAddNewerVersionsBetweenTwoVectors(t *testing.T) {
+	testCases := []struct {
+		name          string
+		localInput    []string
+		incomingInput []string
+		expectedInput []string
+	}{
+		{
+			name:          "testcase1",
+			localInput:    []string{"abc@15"},
+			incomingInput: []string{"def@25", "abc@20"},
+			expectedInput: []string{"def@25", "abc@20"},
+		},
+		{
+			name:          "testcase2",
+			localInput:    []string{"abc@15", "def@30"},
+			incomingInput: []string{"def@35", "abc@15"},
+			expectedInput: []string{"def@35", "abc@15"},
+		},
+		{
+			name:          "testcase3",
+			localInput:    []string{"abc@17", "def@30"},
+			incomingInput: []string{"def@35", "abc@15"},
+			expectedInput: []string{"def@35", "abc@17"},
+		},
+		{
+			name:          "testcase4",
+			localInput:    []string{"abc@20", "ghi@9"},
+			incomingInput: []string{"def@15", "abc@17"},
+			expectedInput: []string{"def@15", "ghi@9", "abc@20"},
```

Comment:
The local and incoming are conflicting HLVs here, as neither HLV dominates the other, so it's not really an expected use case for AddNewerVersions.  It might not need to be part of this PR, but I think a similar unit test that runs both IsInConflict and then AddNewerVersions (when !IsInConflict) would be a good idea.